### ISSUE TITLE
Make static files configurable from admin portal or blacklist certain types by default

### ISSUE THREAD

**Issue by rolandh** [nopSolutions/nopCommerce#2683]
NopCommonStartup.cs has the following code

```
            application.UseStaticFiles(new StaticFileOptions
            {
                FileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), @"Plugins")),
                RequestPath = new PathString("/Plugins"),
                OnPrepareResponse = ctx =>
                {
                    if (!string.IsNullOrEmpty(nopConfig.StaticFilesCacheControl))
                        ctx.Context.Response.Headers.Append(HeaderNames.CacheControl, nopConfig.StaticFilesCacheControl);
                }
            });
```

Now this code seems a bit dangerous as people may include static files unknowingly in their plugin folder. Eg I stored a .pfx key which I used for xero integration. This file was hosted by default in my development environment. Had I not done some security checks I would not have seen this and  store.com/plugins/pluginname/privatekey.pfx could be downloaded.

Can we add the ability to blacklist certain file extensions from the plugins folder? Eg I have added the following code as a temporary workaround. We could easily store these extensions somewhere else, or even simply hard code the extensions of files which would be dubious to host.

            var provider2 = new FileExtensionContentTypeProvider();
            provider2.Mappings.Remove(".pfx");
            provider2.Mappings.Remove(".cs");          //This is not included by default
            provider2.Mappings.Remove(".cshtml");  //This is not included by default
            provider2.Mappings.Remove(".dll");         //This is not included by default
            provider2.Mappings.Remove(".config");   //This is not included by default
            provider2.Mappings.Remove(".db");
            provider2.Mappings.Remove(".old");
            provider2.Mappings.Remove(".ps1");
            provider2.Mappings.Remove(".txt");
            provider2.Mappings.Remove(".json");

            application.UseStaticFiles(new StaticFileOptions
            {
                FileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), @"Plugins")),
                RequestPath = new PathString("/Plugins"),
                ContentTypeProvider = provider2,
                OnPrepareResponse = ctx =>
                {
                    if (!string.IsNullOrEmpty(nopConfig.StaticFilesCacheControl))
                        ctx.Context.Response.Headers.Append(HeaderNames.CacheControl, nopConfig.StaticFilesCacheControl);
                }
            });

edit: I am using the latest develop branch commit of 4.00

**Comment by skoshelev** [comment]
Thanks for your help, I've implemented this feature
Closed #2683

**Comment by rolandh** [comment]
The level of service your provide for what is generally a free product is amazing. Thank you for the quick turn around.
